mm2src/coins/eth.rs

```diff
@@ -2879,30 +2886,42 @@ impl EthCoin {
                     Token::Address(*token_addr),
                     Token::Address(receiver_addr),
                     Token::FixedBytes(secret_hash),
-                    Token::Uint(U256::from(args.time_lock))
+                    Token::Uint(time_lock)
                 ]));
-                let value = U256::from(args.watcher_reward.unwrap_or(0));
+                // wait for the approval tx confirmation only half the time required for the actual htlc tx confirmation
+                let wait_for_approval_confirmation_until = args.wait_for_confirmation_until / 2;
```

Comment:
We could put some random limit on it, but I think we can leave it this way. If the transaction is sent successfully with enough gas and correct addresses it shouldn't fail. Are there any other reasons why it can fail?

Reply:
> but I think we can leave it this way

Yeah, I think so too.

> Are there any other reasons why it can fail?

There is a more general problem with block reorganization specially with polygon, but this should be solved using a higher value for required confs for both taker and maker and if the approve transaction disappears, dependant transactions will fail too, so I see no problem with the current approach.